examples/colormap_normalization/colormap_normalizations_piecewisenorm.py

```diff
@@ -0,0 +1,103 @@
+"""
+============================================
+Examples of arbitrary colormap normalization
+============================================
+
+Here I plot an image array with data spanning for a large dynamic range,
+using different normalizations. Look at how each of them enhances
+different features.
+
+"""
+
+
+import matplotlib.cm as cm
+import matplotlib.colors as colors
+import matplotlib.pyplot as plt
+
+from mpl_toolkits.mplot3d import Axes3D
+
+import numpy as np
+
+from sampledata import PiecewiseNormData
+
+X, Y, data = PiecewiseNormData()
+cmap = cm.spectral
+
+# Creating functions for plotting
+
+
+def make_plot(norm, label=''):
```

Comment:
> What about if you zip it down? (And I think you can have a 100x100 array or something)

A 100x100 array (which already looks very pixelated), with a float16, zipped, is still 12 KB, and contains much less meta information than having the script itself, in case it is later decided to show the example in a higher resolution format.

> Can you address 2 in the docstring/a short comment?
Well of course the documentation explains what it does, but an image is worth a thousand words, and there is nothing like the line plots to understand what is going on. Plus some users may also decide to plot that along there normalised plots to explain the normalization.

> I think 3 is addressed with the linear normalization vs. fancy normalization plots.
Yes, but only if the fake array is cooked carefully for this purpose.

I guess my general point, is that, if it took me making the plots above to convince people people here that what I am adding in this PR, then the users would need something similar to understand what is this for.

I think I will give a try at generating some examples following some of what we discussed, when I get home after work in a couple of hours, and maybe we can go through further rounds of feedback :)